Sources/DeepLearning/Layer.swift

```diff
@@ -1404,12 +1409,75 @@ public struct LSTMCell<Scalar: TensorFlowFloatingPoint>: RNNCell, VectorNumeric
     }
 }
 
+/// A gated recurrent unit (GRU) cell.
+///
+/// Reference: [Gated recurrent unit](https://en.wikipedia.org/wiki/Gated_recurrent_unit).
+@_fixed_layout
+public struct GRUCell<Scalar: TensorFlowFloatingPoint>: RNNCell {
+    public var resetWeight, updateWeight, inputHiddenWeight, hiddenHiddenWeight: Tensor<Scalar>
+    public var resetBias, updateBias, inputHiddenBias, hiddenHiddenBias: Tensor<Scalar>
+
+    public typealias TimeStepInput = Tensor<Scalar>
+    public typealias TimeStepOutput = Tensor<Scalar>
+    public typealias State = Tensor<Scalar>
+    public typealias Input = RNNCellInput<TimeStepInput, State>
+    public typealias Output = RNNCellOutput<TimeStepOutput, State>
+
+    @noDerivative
+    public var stateShape: TensorShape {
+        return TensorShape([1, resetWeight.shape[1]])
+    }
+
+    public var zeroState: State {
+        return Tensor(zeros: stateShape)
+    }
+
+    /// Creates a `GRUCell` with the specified input size and hidden state size.
+    ///
+    /// - Parameters:
+    ///   - inputSize: The number of features in 2-D input tensors.
+    ///   - hiddenSize: The number of features in 2-D hidden states.
+    ///   - seed: The random seed for initialization. The default value is random.
+    public init(inputSize: Int, hiddenSize: Int,
+                seed: (Int64, Int64) = (Int64.random(in: Int64.min..<Int64.max),
+                                        Int64.random(in: Int64.min..<Int64.max))) {
+        let concatenatedHiddenSize = inputSize + hiddenSize
+        let weightShape: TensorShape = [concatenatedHiddenSize, hiddenSize]
+        let biasShape: TensorShape = [hiddenSize]
+        self.resetWeight = Tensor(glorotUniform: weightShape, seed: seed)
```

Comment:
It would be nice if we find ways to expose the initialization of the model parameters. Maybe by having additional initializers that allow you to provide initial values for all parameters. Or even by adding these as arguments to this initializer (e.g., `initialResetWeight: Tensor<Scalar>? = nil`).

Reply:
Also, I find it awkward that sometimes we use a single `UInt64` for seeds and sometimes pairs of two `Int64`s (that represent TensorFlow graph- and op-level seeds I assume). Why don't we just stick to a single `UInt64` everywhere to be consistent and simplify the API?

Reply:
@eaplatanios i agree with the initializers and having more options, i think that's not just in this layer but something to be done for all layers. but as of now only glorot has been written. I also opened #111 in that regard.

Reply:
@eaplatanios i agree with the initializers and having more options, i think that's not just in this layer but something to be done for all layers. but as of now only glorot has been written. I also opened #111 in that regard.